### PR TITLE
[SPARK-35735][SQL][FOLLOWUP] Fix case minute to second regex can cover by hour to minute and unit case-sensitive issue

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
When cast `10:10` to interval minute to second,  it can be catch by hour to minute regex, here to fix this.

### Why are the changes needed?
Fix bug

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Added UT

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala [hunk 1]
```diff
@@ -327,8 +327,8 @@ object IntervalUtils {
         toDTInterval(day, hour, minute, secondAndMicro(second, micro),
           finalSign(firstSign, secondSign))
 
-      case hourMinuteRegex(sign, hour, minute) =>
-        checkDTIntervalStringDataType(DT(DT.HOUR, DT.MINUTE))
+      case hourMinuteRegex(sign, hour, minute)
+        if startField == DT.HOUR && endField == DT.MINUTE =>
```
Comment (AngersZhuuuu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:331]: How about to make all place like this? FYI @cloud-fan @MaxGekk

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:331]: SGTM

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala [hunk 2]
```diff
@@ -369,18 +367,16 @@ object IntervalUtils {
       case dayTimeIndividualLiteralRegex(firstSign, secondSign, value, suffix, unit) =>
         safeToInterval("day-time") {
           val sign = finalSign(firstSign, secondSign)
-          unit match {
-            case "DAY" if suffix == null && value.length <= 9 =>
-              checkDTIntervalStringDataType(DT(DT.DAY, DT.DAY))
+          unit.toUpperCase(Locale.ROOT) match {
```
Comment (AngersZhuuuu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:370]: Here fix a bug about parse `INTERVAL '111' day` failed issue

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master/3.2.
Thank you, @AngersZhuuuu and @cloud-fan for review.

**Pull request closed without merging**